checker/examples/units-extension/FrequencyRelations.java

```diff
@@ -35,11 +39,19 @@ public UnitsRelations init(ProcessingEnvironment env) {
     // Division of a scalar by seconds yields Hertz.
     // Other divisions yield an unannotated value.
     public /*@Nullable*/ AnnotationMirror division(AnnotatedTypeMirror p1, AnnotatedTypeMirror p2) {
-        if (p1.getAnnotations().isEmpty() && p2.getAnnotations().contains(s)) {
+        if (noUnits(p1) && AnnotationUtils.containsSame(p2.getAnnotations(), s)) {
             return hz;
         }
 
         return null;
     }
 
-}
+    // returns true if the type mirror either has no annotations or has only 1 annotation and it is UnknownUnits
+    // any number that isn't assigned a unit will automatically gets the annotation UnknownUnits, eg: int x = 5; //x has @UnknownUnits
+    private boolean noUnits(AnnotatedTypeMirror t) {
+        Set<AnnotationMirror> annos = t.getAnnotations();
+        return annos.isEmpty() ||
+                (annos.size() == 1 &&
```

Comment:
Don't you have the guarantee that you always have at least one annotation?
Why is annos.isEmpty a possibility?